src/do_unmount/do_unmount.c

```diff
@@ -0,0 +1,382 @@
+        }
+    }
+
+    /* Close file and free memory */
+    fclose(fp);
+    if (line)
+        free(line);
+    if (path)
+        free(path);
+    #endif
+}
+
+/* Pass arguments to mountinfo and store output in a list of paths to unmount */
+static int
+populate_unmount_list(RC_STRINGLIST **list, int argc, char **argv)
+{
+    int size = 0;       /* Number of paths to unmount */
+    FILE *fp;           /* File pointer to the output of the command */
+    char *path = NULL;  /* Path to add to the list */
+    size_t len = 0;     /* Length of the line read */
+
+
+    /* Open the command for reading */
+    fp = popen_vec("mountinfo", argc-2, argv+2);
+    if (fp == NULL) {
+        printf("Failed to run mountinfo command, can't unmount anything!\n");
+        exit(1);
+    }
+
+    *list = rc_stringlist_new();
+    /* Read the output a line at a time */
+    while (getline(&path, &len, fp) != -1) {
+        path[strlen(path) - 1] = '\0'; /* Remove trailing '\n' */
+        if (strstr(path, "/proc") == NULL) {
+        rc_stringlist_add(*list, path);
+        size++;
+        }
+    }
+
+    pclose(fp);
+    if (path)
+        free(path);
+    return size;
+}
+
+/* If unmount command fails terminate the programs using it and retry */
+static int
+unmount_with_retries(char *command, char *mount_point)
+{
+    int retry = 4;                                      /* Effectively TERM, sleep 1, TERM, sleep 1, KILL, sleep 1 */
+    size_t len = 0;                                     /* Length of the line read */
+    char *pids = NULL;                                  /* Line read from the fuser command */
+    FILE *fp = NULL;                                    /* File pointer to the output of the command */
+    char *fuser_command;                                /* Command to execute fuser and find the pids of the processes that use the mount point */
+    char *kill_command;                                 /* Command to kill the processes that use the mount point */
+    char pidString[8];                                  /* String to store the pid (max value is 4194304 source https://unix.stackexchange.com/questions/16883/what-is-the-maximum-value-of-the-process-id) */
+    outcome_t retVal = SUCCESS;                         /* Return value of the function */
+    const char *f_opts = "-m -c";                       /* fuser options */
+    const char *f_kill = "-s ";                         /* fuser kill options */
+    const char* timeout = getenv("rc_fuser_timeout");   /* fuser timeout */
+
+    #ifdef __linux__
+    f_opts = "-m";
+    f_kill = "-";
+    #endif
+
+    /* Default timeout value if environment variable is not set */
+    if (timeout == NULL)
+        timeout = "60";
+
+    /* Allocate memory for the commands */
+    xasprintf(&fuser_command, "timeout -s KILL %s fuser %s %s 2>/dev/null", timeout, f_opts, mount_point);
+    xasprintf(&kill_command, "fuser %sTERM -k %s \"%s\" >/dev/null 2>&1", f_kill, f_opts, mount_point);
```

Comment:
Same issue with non-robust escape and potential random code execution here. Use fork+exec or posix_spawn.